osquery/tables/system/darwin/extended_attributes.cpp

```diff
@@ -164,72 +130,87 @@ void parseWhereFrom(QueryData &results,
 
   for (CFIndex i = 0; i < count; i++) {
     CFStringRef attribute = (CFStringRef)CFArrayGetValueAtIndex(attribs, i);
-    Row r;
-    r["path"] = path;
-    r["directory"] = directory;
-    r["key"] = "where_from";
-    r["value"] = stringFromCFString(attribute);
-    r["base64"] = INTEGER(0);
-    results.push_back(r);
+    auto where_from_attribute = stringFromCFString(attribute);
+    if (!where_from_attribute.empty()) {
+      setRow(results, path, "where_from", where_from_attribute);
+    }
   }
 
   CFRelease(attributes);
 }
 
-// Parse a given XAttrAttribute struct as an OS X quarantine string
-void parseQuarantineFile(QueryData &results,
-                         const XAttrAttribute &x_att,
-                         const std::string &path,
-                         const std::string &directory) {
-  std::vector<std::string> values = osquery::split(x_att.attribute_data, ";");
-  if (values.size() < 2) {
+void extractQuarantineProperty(const std::string table_key_name,
+                               CFTypeRef property,
+                               const std::string path,
+                               QueryData &results) {
+  std::string value;
+  if (CFGetTypeID(property) == CFStringGetTypeID()) {
+    value = stringFromCFString((CFStringRef)property);
+  } else if (CFGetTypeID(property) == CFDateGetTypeID()) {
+    auto unix_time = CFDateGetAbsoluteTime((CFDateRef)property) +
+                     kCFAbsoluteTimeIntervalSince1970;
+    value = INTEGER(std::llround(unix_time));
+  } else if (CFGetTypeID(property) == CFURLGetTypeID()) {
+    value = stringFromCFString(CFURLGetString((CFURLRef)property));
+  }
+  setRow(results, path, table_key_name, value);
+}
+
+void parseQuarantineFile(QueryData &results, const std::string &path) {
+  CFURLRef url = CFURLCreateFromFileSystemRepresentation(
```

Comment:
Yup, missed that one while moving code around. Good catch.
From Apple's `CFURL` docs:
> If a function fails to create a CFURL object, it returns NULL, which you must be prepared to handle.